projects/mercury/src/metadata/common/vocabularyUtils.js

```diff
@@ -215,6 +222,40 @@ const generatePropertyEntry = (vocabulary, predicate, shape) => {
     };
 };
 
+/**
+ * Returns list of
+ * @param hierarchyClass
+ * @returns {*[]|*}
+ */
+const getAllowedDescendantTypes = (hierarchyClass) : string[] => {
+    const descendants = getFirstPredicateList(hierarchyClass, constants.HIERARCHY_DESCENDANTS);
+    if (descendants && descendants.length > 0)
+        return getFirstPredicateList(hierarchyClass, constants.HIERARCHY_DESCENDANTS).map(p => p['@id']);
+    return [];
+}
+
+/**
+ * Determines a hierarchy between directory types that are a part of file browser structure.
+ *
+ * @param vocabulary
+ * @returns {HierarchyLevel[]}    An array of hierarchy level objects.
+ */
+export const determineHierarchy = (vocabulary): HierarchyLevel[] => {
+    const hierarchyClasses = getClassesInCatalog(vocabulary)
+        .filter(entry => getFirstPredicateValue(entry, constants.IS_PART_OF_HIERARCHY) === true);
```

Comment:
=== true is redundant, or not in JS?

Reply:
Here it's better to be precise - it doesn't have to be boolean (types in js are not enforced), so js will accept and find any value that is different than `null`, `undefined` or `false`.